tests/sim_asyncio_ioc_override.py

```diff
@@ -0,0 +1,33 @@
+from argparse import ArgumentParser
+
+import asyncio
+import os
+import re
+from pathlib import Path
+
+from softioc import softioc, builder, asyncio_dispatcher
+
+
+if __name__ == "__main__":
+    # Being run as an IOC, so parse args and set prefix
+    parser = ArgumentParser()
+    parser.add_argument('prefix', help="The PV prefix for the records")
+    parsed_args = parser.parse_args()
+    builder.SetDeviceName(parsed_args.prefix)
+
+    # Load the base records without DTYP fields
+    macros = dict(device=parsed_args.prefix)
+    with open(Path(__file__).parent / "hw_records.db") as f:
+        for line in f.readlines():
+            if not re.match(r"\s*field\s*\(\s*DTYP", line):
+                builder.AddDatabaseLine(line, macros)
+
+    # Override DTYPE and OUT, and provide a callback
+    gain = builder.boolOut("GAIN", on_update=print)
+    softioc.devIocStats(parsed_args.prefix)
+
+    # Run the IOC
+    builder.LoadDatabase()
+    event_loop = asyncio.get_event_loop()
+    softioc.iocInit(asyncio_dispatcher.AsyncioDispatcher(event_loop))
+    event_loop.run_forever()
```

Comment:
@Araneidae I wanted to see if you thought this was the best approach to solving the "load existing database, strip out DTYP, then load some pythonSoftIOC override records on top"

Reply:
Not a huge fan.  Building a database up line by line is very error prone, there's nothing stopping you from creating a horribly malformed file this way.

This is, essentially a hack.  As such I'd suggesting being a bit more honest about the hack and not add half-baked methods to the API.  In this particular case what it looks like you want to do is:

- Process `hw_records.db` to remove all lines containing any `field DTYP` definitions.  Put the result in a temporary file (just like `LoadDatabase` does under the hood...).
- Use `dbLoadDatabase` to load the hacked file and apply any MSI style substitutions.

Reply:
This sounds like a better solution, I'll try it out